File: java/src/main/java/it/giannibombelli/wsc2026/giftcard/application/commands/ConfirmTopUp.java
package it.giannibombelli.wsc2026.giftcard.application.commands;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.application.Command;
import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.giftcard.domain.giftcard.GiftCardId;

public record ConfirmTopUp(GiftCardId aggregateId, Money amount) implements Command<GiftCardId> {
    public ConfirmTopUp {
        Require.requireArgument(aggregateId, "giftCardId");
        Require.requireArgument(amount, "amount");
    }
}

File: typescript/src/giftcard/application/commands/confirmTopUp.ts
// Command per confermare una ricarica sulla gift card.

import type { Command } from '@/common/application/command.js';
import type { Money } from '@/common/domain/primitive/money.js';
import type { GiftCardId } from '../../domain/giftcard/giftCardId.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type ConfirmTopUp = Command<'ConfirmTopUp', GiftCardId> & {
  readonly amount: Money;
};

export function confirmTopUp(aggregateId: GiftCardId, amount: Money): ConfirmTopUp {
  requireArgument(aggregateId, 'giftCardId');
  requireArgument(amount, 'amount');
  return { kind: 'ConfirmTopUp', aggregateId, amount };
}
